lib/buffer.js

```diff
@@ -104,6 +155,9 @@ function fromString(string, encoding) {
   if (typeof encoding !== 'string' || encoding === '')
     encoding = 'utf8';
 
+  if (!Buffer.isEncoding(encoding))
+    throw new TypeError('"encoding" must be a valid string encoding');
+
```

Comment:
This is also part of the semver-major API change but represents a new throw and can be pulled back out if necessary.